dawn/renderer/components/DNav.js

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import {
+  Navbar,
+  ButtonToolbar,
+  ButtonGroup } from 'react-bootstrap';
+import { remote } from 'electron';
+import smalltalk from 'smalltalk';
+import ConfigBox from './ConfigBox';
+import UpdateBox from './UpdateBox';
+import StatusLabel from './StatusLabel';
+import TooltipButton from './TooltipButton';
+
+const storage = remote.require('electron-json-storage');
+
+class DNav extends React.Component {
+  static saveAddress(currentAddress) {
+    const prompt = smalltalk.prompt(
+      'Enter the IP address of the robot:',
+      'Examples: 192.168.13.100, 127.0.0.1',
+      currentAddress,
+    );
+    prompt.then((value) => {
+      if (/\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}/.test(value)) {
```

Comment:
There's already a `getValidationState` (yes, this should be given a better name like `validateIP`) in renderer/utils/utils.js which has a more precise regex.